src/util/rc.js

```diff
@@ -55,13 +26,42 @@ export function findRc(name: string, parser: Function): Object {
     addConfigPath(home, `.${name}rc`);
   }
 
-  addRecursiveConfigPath(process.cwd(), `.${name}rc`);
+  // add .yarnrc locations relative to the cwd
+  while (true) {
+    configPaths.unshift(path.join(cwd, `.${name}rc`));
+
+    const upperCwd = path.dirname(cwd);
```

Comment:
Not really, because if we were to do that we wouldn't read the yarnrc files in $HOME, for example.